Assets/Scripts/Player/PlayerController.cs

```diff
@@ -218,36 +223,50 @@ public void OnLook(InputAction.CallbackContext value)
 
 				transform.rotation *= Quaternion.AngleAxis(rot.x * _horizontalLookMultiplier, Vector3.up);
 
-				_headRotation += rot.y * _verticalLookMultiplier;
+				_headRotation -= rot.y * _verticalLookMultiplier;
 				_headRotation = Mathf.Clamp(_headRotation, -89, 89);
 				_head.transform.localRotation = Quaternion.AngleAxis(_headRotation, Vector3.right);
 			}
 		}
 
 		public void OnShoot(InputAction.CallbackContext value)
 		{
-			if (value.phase == InputActionPhase.Performed && // Already done the action when the button is pressed
-				!IsGamePaused() && // Don't shoot if the game is paused
-				!_isReloading && // We can't do it if we are reloading
-				_slotWeapons[(int)_currentWeapon].AmmoInGun > 0 && // We can't do it if there are no ammo left
-				_carriedObject == null) // Can't do anything while we are carrying an object
```

Comment:
Where did you move these checks?

Reply:
I moved the responsibility of the ammo counter and reloading time to the gun (or "held item"), and it is check by the PlayerController via `CanHandlePrimaryAction`.

Reply:
held item isn't the gun
It's when you are carrying an object (like a sentry)